pkg/replica/replica.go

```diff
@@ -1156,3 +1177,72 @@ func (r *Replica) GetReplicaStat() (int64, int64) {
 
 	return lastModifyTime, headFileSize
 }
+
+func (r *Replica) GetFileInterval(includeBackingFileLayer bool) (sparse.GetFileInterval, error) {
+	if err := preload(&r.volume, includeBackingFileLayer); err != nil {
+		return nil, err
+	}
+	// baseDiskIndex is the smallest index in r.volume.files that is needed to export
+	// Note that index 0 in r.volume.files is nil. Index 1 is backing file if the volume has backing file.
+	// Otherwise, index 1 is regular disk file.
+	baseDiskIndex := 1
+
+	getFileInterval := func(ctx context.Context) (<-chan sparse.FileInterval, <-chan error, error) {
+		fileIntervalChannel := make(chan sparse.FileInterval)
+		errChannel := make(chan error)
+
+		go func() {
+			defer close(fileIntervalChannel)
+			defer close(errChannel)
+
+			// The replica consists of data and and hole region.
+			// Inside this function scope, let represent 0 as hole sector and 1 as data sector.
+			// Then a replica could look like 010001111...
+
+			getSectorType := func(diskIndex byte) int {
+				if diskIndex >= byte(baseDiskIndex) {
+					return 1
```

Comment:
Use const to represent data or holes. e.g.:
```
const (
    sectorTypeData = 1  // The value is consist with sparse.SparseData
    sectorTypeHole = 2
)
```

Or you may directly use `sparse.SparseHole` or `sparse.SparseData`

Reply:
Thanks! I added the const declaration as suggested. But I still prefer 0 as hole and 1 as data. This shouldn't affect anything else because the convention is only within the scope of this function.

Reply:
Sure. The value is not required to be the consist with `sparse.SparseData` or `sparse.SparseHole`